homeassistant/components/octoprint/config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Config flow for OctoPrint integration."""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_HOST, CONF_NAME, CONF_SENSORS
+import homeassistant.helpers.config_validation as cv
+
+from .const import (  # pylint:disable=unused-import
+    CONF_BED,
+    CONF_NUMBER_OF_TOOLS,
+    DOMAIN,
+)
+from .octoprintapi import OctoPrintAPI
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "OctoPrint"
+
+SENSORS = [
+    "Printing",
+    "Printing Error",
+    "Temperatures",
+    "Current State",
+    "Job Percentage",
+    "Time Remaining",
+    "Time Elapsed",
+]
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NUMBER_OF_TOOLS, default=0): cv.positive_int,
+        vol.Optional(CONF_BED, default=False): cv.boolean,
```

Comment:
These can be fetched from OctoPrint and should not be part of the flow or data set in Home Assistant.

Reply:
Removed